File: 485_max_consecutive_ones/Solution.go

package main

func findMaxConsecutiveOnes(nums []int) int {
    var maxConsecutiveResult = 0;
	var maxConsecutiveNow = 0;
	for i := 0; i < len(nums); i++ {
		if (nums[i] == 1) {
			maxConsecutiveNow++;
		} else {
			if maxConsecutiveNow > maxConsecutiveResult {
				maxConsecutiveResult = maxConsecutiveNow;
			}
			maxConsecutiveNow = 0;
		}
	}

	if maxConsecutiveNow >= maxConsecutiveResult {
		return maxConsecutiveNow;
	} 
	return maxConsecutiveResult;
}


func main() {
	var nums = []int{1, 1, 0, 1, 1, 1}
	var result = findMaxConsecutiveOnes(nums)
	println(result)
}

File: 485_max_consecutive_ones/Solution.java
public class Solution {

    public static void main(String[] args) {

        int[] nums = { 1, 1, 0, 1, 1, 1 };
        System.out.println(findMaxConsecutiveOnes(nums));

        nums = new int[] { 1, 0, 1, 1, 0, 1 };
        System.out.println(findMaxConsecutiveOnes(nums));
    }

    public static int findMaxConsecutiveOnes(int[] nums) {

        int maxConsecutiveResult = 0;
        int maxConsecutiveNow = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 1) {
                maxConsecutiveNow++;
            } else {
                maxConsecutiveResult = (maxConsecutiveNow >= maxConsecutiveResult) ? maxConsecutiveNow
                        : maxConsecutiveResult;
                maxConsecutiveNow = 0;
            }
        }

        return (maxConsecutiveNow >= maxConsecutiveResult) ? maxConsecutiveNow
        : maxConsecutiveResult;
    }

}